mantidproject/mantid
# Fit Script Generator - Python Script for Simultaneous Fitting

**Description of work.**
This PR adds the ability to generate a python script used for simultaneous fitting. It also connects this new ability to the FitScriptGenerator interface.

**To do:**
- [ ] Test to run the generated Sequential and Simultaneous scripts to make sure they don't error
- [ ] Fit Script Generator documentation page. Link to this page where appropriate.
- [ ] Update existing tests

**To test:**
1. Comment out this line
https://github.com/mantidproject/mantid/blob/5077722a251590b9196563d8d23245ce15e04699/scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_controls_view.py#L28
1. Open Muon Analysis interface
2. Load MUSR 62260
3. Go to Fitting tab and click `Fit Generator`
4. When the `Fit Script Generator` opens, there should be 4 domains loaded into the interface
5. Change the Fitting Mode to `Simultaneous` (bottom right)
6. Look at the top right and make sure `All Domains` is selected.
7. Right click the area under the combo box and add a `GausOsc` function.
8. Change `Frequency` to have value `1.3`, and make it Global
9. Now click `Generate Script to Clipboard`
10. Open a new python script window on Workbench, and make sure it is empty. Right click and paste the generated code.
11. Run this python script and wait. You should get four plots representing a simultaneous fit:

![image](https://user-images.githubusercontent.com/40830825/120819809-d7e10700-c54b-11eb-8b41-e584e97868cf.png)


Fixes #29913

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
